### ISSUE TITLE
[WIN32] Should not force include algorithm and compile as cpp.

### ISSUE THREAD

**Issue by xpol** [cocos2d/cocos2d-x#12704]
in file https://github.com/cocos2d/cocos2d-x/blob/v3/templates/lua-template-default/frameworks/runtime-src/proj.win32/HelloLua.vcxproj .

Should not force include `algorithm` and compile as cpp.

```
  <ForcedIncludeFiles>algorithm</ForcedIncludeFiles>
  <CompileAs>CompileAsCpp</CompileAs>
```

This is not friendly for c source code.

**Comment by minggo** [comment]
> This is not friendly for c source code.

Could you tell me why? And what's your suggestion?

**Comment by xpol** [comment]
Just removed the 2 lines and it works fine.
(Its too long, can't remember the detals.)

**Comment by minggo** [comment]
Ok, i will do more research.

**Comment by minggo** [comment]
fixed in #16195
